plaso/parsers/powershell_transcript.py

```diff
@@ -0,0 +1,343 @@
+# -*- coding: utf-8 -*-
+"""Parser for powershell transcript files."""
+
+import copy
+import pyparsing
+
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import definitions
+from plaso.parsers import text_parser
+from plaso.parsers import manager
+
+
+class PowerShellTranscriptEventData(events.EventData):
+  """PowerShell transcript event data.
+
+  Attributes:
+    build_version (str): Build number of current version.
+    clr_version (str): Common Language Runtime version.
+    command (str): Command that was executed.
+    compatible_versions (str): Compatible PowerShell versions.
+    config_name (str): Configuration name.
+    edition (str): PowerShell edition
+    host_application (str): Application that executed command.
+    machine (str): Hostname of machine.
+    process_id (str): Process ID.
+    remoting_protocol_version (str): PowerShell
+      remote management protocol version.
+    runas_user (str): User context of execution.
+    serialization_version (str): Serialization method version.
+    username (str): User that executed command.
+    version (str): PowerShell version.
+    ws_man_stack_version (str): WS-Management stack version
+    username (str): User that executed command.
+  """
+
+  DATA_TYPE = 'powershell:transcript:event'
+
+  def __init__(self):
+    """Initializes event data."""
+
+    super(PowerShellTranscriptEventData,
+      self).__init__(data_type=self.DATA_TYPE)
+    self.build_version = None
+    self.clr_version = None
+    self.command = None
+    self.compatible_versions = None
+    self.config_name = None
+    self.edition = None
+    self.host_application = None
+    self.machine = None
+    self.process_id = None
+    self.remoting_protocol_version = None
+    self.runas_user = None
+    self.serialization_version = None
+    self.username = None
+    self.version = None
+    self.ws_man_stack_version = None
+
+class PowerShellTranscriptParser(text_parser.PyparsingMultiLineTextParser):
+  """Parses events from PowerShell transcript files."""
+
+  NAME = 'powershell_transcript'
+
+  DATA_FORMAT = 'PowerShell transcript event'
+
+  # PowerShell transcript lines can be very long.
+  MAX_LINE_LENGTH = 65536
+
+  _SEPARATOR = '**********************'
+
+  _COLON = pyparsing.Literal(': ').suppress()
+
+  # key always start with an uppercase character
+  _METADATA_KEY = (pyparsing.Word(pyparsing.alphas.upper(), ' ' +
+    pyparsing.alphas) + _COLON)
+  # negative lookahead needed for correct identification
+  _METADATA_VALUE = (~_METADATA_KEY +
+    pyparsing.Word(pyparsing.printables, ' ' + pyparsing.printables))
+
+  _METADATA_LINE = (
+    _METADATA_KEY +
```

Comment:
style nit - in these cases where you have an open parenthesis, use a hanging indent of 4 spaces